captum/optim/_utils/atlas.py

```diff
@@ -0,0 +1,250 @@
+            of: [n_samples, n_channels].
+        grid_size (Tuple[int, int]): The size of grid cells to use. The grid_size
+            variable should be in the format of: [width, height].
+        min_density (int, optional): The minimum number of points for a cell to be
+            counted.
+        normalize (bool, optional): Whether or not to remove outliers from an xy
+            coordinate grid tensor, and rescale it to [0, 1].
+        x_extent (Tuple[float, float], optional): The x axis range to use.
+        y_extent (Tuple[float, float], optional): The y axis range to use.
+
+    Returns:
+        grid_vecs (torch.tensor): A tensor containing all the direction vector that
+            were created, stacked along the batch dimension.
+        cell_coords (list of Tuple[int, int, int]): List of coordinates for grid
+            spatial positions of each direction vector, and the number of samples used
+            for the cell. The list for each cell is in the format of:
+            [x_coord, y_coord, number_of_samples_used].
+    """
+
+    assert xy_grid.dim() == 2 and xy_grid.size(1) == 2
+    assert raw_activations.dim() == 2
+
+    if normalize:
+        xy_grid = normalize_grid(xy_grid)
+    indices = calc_grid_indices(
+        xy_grid, grid_size, x_extent=x_extent, y_extent=y_extent
+    )
+    grid_vecs, vec_coords = extract_grid_vectors(
+        indices, raw_activations, grid_size, min_density
+    )
+    return grid_vecs, vec_coords
+
+
+def create_atlas(
+    cells: Union[torch.Tensor, List[torch.Tensor]],
+    coords: List[Tuple[int, int]],
+    grid_size: Tuple[int, int],
+    base_tensor: Callable = torch.ones,
+) -> torch.Tensor:
+    """
+    Create an NCHW atlas grid image tensor from a set of NCHW image tensors and their
+    corresponding grid coordinates.
+
+    Args:
+
+        cells (list of tensor or tensor): A list or stack of image tensors made with
+            atlas direction vectors.
+        coords (list of Tuple[int, int] or list of Tuple[int, int, int]): A list of
+            coordinates to use for the atlas image tensors. The first 2 values in each
+            coordinate list should be: [x, y, ...].
+        grid_size (Tuple[int, int]): The size of grid cells to use. The grid_size
+            variable should be in the format of: [width, height].
+        base_tensor (Callable, optional): What to use for the atlas base tensor. Basic
+            choices are: torch.ones or torch.zeros.
+
+    Returns:
+        atlas_canvas (torch.tensor): The full activation atlas visualization.
+    """
+
+    if torch.is_tensor(cells):
+        assert cast(torch.Tensor, cells).dim() == 4
+        cells = [c.unsqueeze(0) for c in cells]
+
+    assert len(cells) == len(coords)
+    assert all([c.shape == cells[0].shape for c in cells])
+    assert all([c.device == cells[0].device for c in cells])
+    assert cells[0].dim() == 4
+
+    cell_b, cell_c, cell_h, cell_w = cells[0].shape

```

Comment:
Thank you for the explanation, @ProGamerGov! I only meant to comment in the code that  
```
cell_b -> number of images
cell_c -> image channel
cell_h ->  image hight
cell_w -> image width
```

Also since this is image specific it would make sense to move atlas.py under `_utils/image`.

Reply:
@NarineK Ah, okay I've added the comments for cell dimensionality!